pkg/operator/csi/csidrivernodeservicecontroller/csi_driver_node_service_controller.go

```diff
@@ -227,3 +258,25 @@ func replacePlaceholders(manifest []byte, spec *opv1.OperatorSpec) []byte {
 	replaced := strings.NewReplacer(pairs...).Replace(string(manifest))
 	return []byte(replaced)
 }
+
+func (c *CSIDriverNodeServiceController) syncDeleting(ctx context.Context, opSpec *opv1.OperatorSpec, opStatus *opv1.OperatorStatus, syncContext factory.SyncContext) error {
+	klog.V(4).Infof("syncDeleting")
+	required, err := c.getDaemonSet(opSpec)
+	if err != nil {
+		return err
+	}
+
+	err = c.kubeClient.AppsV1().DaemonSets(required.Namespace).Delete(ctx, required.Name, metav1.DeleteOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	} else {
+		klog.V(2).Infof("Deleted DaemonSet %s/%s", required.Namespace, required.Name)
+	}
+
+	// All removed, remove the finalizer as the last step
```

Comment:
is the removal pattern actually deleting the daemonset or do we want to remove the entire namespace?

Reply:
Deleting the daemonset only.